src/openloco/window.cpp

```diff
@@ -130,13 +130,35 @@ namespace openloco::ui
         *outZ = regs.dx;
     }
 
+    // 0x004C641F
+    // regs.dl:  underground
+    // regs.esi: w
+    // regs.edi: vp
     static void viewport_set_underground_flag(bool underground, ui::window* w, ui::viewport* vp)
     {
-        registers regs;
-        regs.esi = (int32_t)w;
-        regs.edi = (int32_t)vp;
-        regs.dl = underground;
-        call(0x4C641F, regs);
+        if (w->type != WindowType::main)
```

Comment:
This may be down to personal preference, but I think it's better to invert this and return early. That way, you don't have to encapsulate the entire body in this if block.